File: tidyStops.R
# tidyStops.R is a function that cleans data from the Stop-Extract.csv file.

tidyStops <- function(){ 
        # Load necessary libraries
        library(dplyr); library(lubridate)
        
        # Read in data and set to local variable
        stops_tbl <- tbl_df(read.csv("~/Washougal-Mill-Study/CSVs/Stops-Extract.csv"))
        
        # Cleans names
        names(stops_tbl) <- tolower(names(stops_tbl))
        names(stops_tbl) <- sub("\\.", "", names(stops_tbl))
        colnames(stops_tbl)[3] <- "loom"
        
        # Set loom as factor variable
        stops_tbl$loom <- as.factor(stops_tbl$loom)
        stops_tbl$date <- mdy(stops_tbl$date)
        
        # Removes records of experimental millstyles
        stops_tbl <- stops_tbl[!grepl("[Ee][Xx]", stops_tbl$range),]
        
        # Creates timing features: total time, time per stop
        stops_tbl <- stops_tbl %>% 
                #removes potential to create inf values
                filter(combinedstops > 0) %>%
                mutate(minutesweaving = (mpx * 1000)/ppm,
                       hoursweaving = minutesweaving/60,
                       timetilstop = minutesweaving/combinedstops)
        
        # Returns table
        return(stops_tbl)
}

File: Python Scripts/tidyStops.py
# tidyStops.py is a function that cleans data from the Stop-Extract.csv file.

def tidyStops():
	import pandas as pd 

	# Read in data and set to local variable
	stops_filename = '~/Desktop/Pendleton/CSVs/Stops-Extract.csv'
	stops_df = pd.read_csv(stops_filename)

	# Clean column names
	stops_df.columns = map(str.lower, stops_df.columns)
	stops_df.columns = stops_df.columns.str.replace(" ", "")

	# Rename column name to 'loom'
	stops_df = stops_df.rename(columns = {'lm#':'loom'})

	# Remove Experimental Values
	stops_df = stops_df[stops_df['range'].str.contains("EX|ex") == False]

	# Set date field to date time data type
	stops_df['date'] = pd.to_datetime(stops_df['date'])

	return stops_df